### ISSUE TITLE
Slow

### ISSUE THREAD

**Issue by chrisbriers** [Q42/Q42.HueApi#36]
Hello

First of all , congrats with this api!

I want to control the Philips Hue from a C# application.
I got it up and running, but the lights are responding versy slow. With the Philips app it works fine.

So here's my code:

        HueClient client = new HueClient("192.168.0.100");

        private void Form1_Load(object sender, EventArgs e)
        {
            
           client.RegisterAsync("test user", "mypersonalappkey");

            client.Initialize("mypersonalappkey");

          
        }

        private void button1_Click(object sender, EventArgs e)
        {
            var command = new LightCommand();
            command.On = true;

            command.TransitionTime = new TimeSpan(0,0,0,0,1);
            command.TurnOn().SetColor("FF00AA");
            //command.TurnOff();

            client.SendCommandAsync(command, new List<string> { "2"});

        }

If I execute the command, it takes sometimes up to 5 seconds for the lights to respond....

So what am I missing?

Thank you for the help!

Chris Briers

**Comment by michielpost** [comment]
You're missing nothing. It shouldn't be slow.
The API uses the same api requests as the apps. So there shouldn't be any difference.

You can try to monitor the request to the bridge with a tool like Fiddler (http://www.telerik.com/fiddler) to see if you can see what's wrong.

**Comment by chrisbriers** [comment]
Thank you for your comment.
I just tried a new network, with a better router and it is better. So I think the issue was my hardware.

No i am programming some cool effect. But how do I use the light command extension class? It is a static class, so i can not create a instance..

Do you have some examples for some cool effects ? 


Thank you

**Comment by paulpdaniels** [comment]
Extension methods in C# can be used like instance methods on the type they extend (look for the first argument preceded by the "this" keyword).

See [here](https://msdn.microsoft.com/en-us/library/bb383977.aspx) for reference.

Essentially though you can do stuff like

```csharp

LightCommand lightCommand = new LightCommand();

//Note these are the methods in LightCommandExtensions 
lightCommand.TurnOn().SetColor(myColor);


```

**Comment by ddivita** [comment]
I think the issue may be around using this library in a web application. I am noticing when trying to send a command i never get a result back or I am not waiting long enough. I did wait over 10 minutes and that is not ideal. Sending the same command in a console application works fine.

**Comment by ddivita** [comment]
Changing my controller methods to <code>async Task<ActionResult></code> allowed the SendCommand to run as expected. I am not sure what is different about the result of that method as opposed to other methods. For example, running <code>DeleteLightAsync(..).Result</code> worked without an issue in a non synchronous method.

**Comment by michielpost** [comment]
I created an MVC website and was able to sort of reproduce this problem. If I was using the default synchronous controllers and used the .Result property I noticed strange behavior on the SendCommand method when the List of lights parameter contained multiple lights.
The library sends these requests in parallel. I did some improvements and got slightly better results. New build is up on NuGet.

However, it is advised to use async controller methods and to await the calls. That way, everything will work as expected.
